ipam/provider/aws.go

```diff
@@ -0,0 +1,256 @@
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			_, err = f.WriteString(values)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+		}
+
+		err = framework.HelmCmd(fmt.Sprintf("registry install quay.io/giantswarm/apiextensions-aws-config-e2e-chart:stable -- -n %s --values %s", awsConfigDeploymentName(clusterName), f.Name()))
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		return nil
+	}
+	b := backoff.NewConstant(framework.ShortMaxWait, framework.ShortMaxInterval)
+	n := func(err error, delay time.Duration) {
+		log.Println("level", "debug", "message", err.Error())
+	}
+
+	err = backoff.RetryNotify(o, b, n)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+func (aws *AWS) DeleteCluster(clusterName string) error {
+	deploymentName := awsConfigDeploymentName(clusterName)
+	err := framework.HelmCmd(fmt.Sprintf("delete --purge %s", deploymentName))
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+func (aws *AWS) installCertResources(clusterName string) error {
+	{
+		aws.logger.Log("level", "debug", "message", "installing cert resource chart")
+
+		o := func() error {
+			err := framework.HelmCmd(fmt.Sprintf("delete --purge %s", certDeploymentName(clusterName)))
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			cmdStr := fmt.Sprintf("registry install quay.io/giantswarm/apiextensions-cert-config-e2e-chart:stable -- -n %s --set commonDomain=${COMMON_DOMAIN} --set clusterName=%s", certDeploymentName(clusterName), clusterName)
+			err = framework.HelmCmd(cmdStr)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			return nil
+		}
+		b := backoff.NewExponential(framework.ShortMaxWait, framework.ShortMaxInterval)
+		n := backoff.NewNotifier(aws.logger, context.Background())
+		err := backoff.RetryNotify(o, b, n)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		aws.logger.Log("level", "debug", "message", "installed cert resource chart")
+	}
+
+	{
+		aws.logger.Log("level", "debug", "message", "waiting for k8s secret to be there")
+
+		o := func() error {
+			n := fmt.Sprintf("%s-api", clusterName)
+			_, err := aws.hostFramework.K8sClient().CoreV1().Secrets("default").Get(n, metav1.GetOptions{})
+			if err != nil {
+				return microerror.Maskf(waitError, "k8s secret is still missing")
+			}
+
+			return nil
+		}
+		b := backoff.NewExponential(framework.ShortMaxWait, framework.ShortMaxInterval)
+		n := func(err error, delay time.Duration) {
+			aws.logger.Log("level", "debug", "message", err.Error())
+		}
+
+		err := backoff.RetryNotify(o, b, n)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		aws.logger.Log("level", "debug", "message", "k8s secret is there")
+	}
+
+	return nil
+}
+
+func awsConfigDeploymentName(clusterName string) string {
+	return fmt.Sprintf("aws-config-e2e-%s", clusterName)
+}
+
+func certDeploymentName(clusterName string) string {
+	return fmt.Sprintf("%s-cert-config-e2e", clusterName)
+}
```

Comment:
Nit: I think it would be nice if both deployment names started or ended with the clusterName.

Reply:
Roger. I'll change.